tests/TestCase/Model/Behavior/SearchBehaviorTest.php

```diff
@@ -186,6 +187,35 @@ public function testFinder()
     }
 
     /**
+     * Test the custom "emptyValues" configuration
+     *
+     * @return void
+     */
+    public function testEmptyValues()
+    {
+        $queryString = [
+            'foo' => 'a',
+            'search' => 'b',
+            'group' => false
+        ];
+
+        $query = $this->Articles->find('search', ['search' => $queryString]);
+        $this->assertSame(2, $query->clause('where')->count());
+
+        $this->Articles->behaviors()->get('Search')->configShallow([
+            'emptyValues' => ['a']
+        ]);
+        $query = $this->Articles->find('search', ['search' => $queryString]);
+        $this->assertSame(2, $query->clause('where')->count());
+
+        $this->Articles->behaviors()->get('Search')->configShallow([
+            'emptyValues' => ['a', false]
```

Comment:
This test is pointless. The `emptyValues` should be overwritten even when initially setting up the behavior using `addBehavior()` which doesn't happen currenlty. For that you will have to add `SearchBehavior::initialize()` and overwrite `emptyValues` if present in passed `$config` array.